future-apps/payroll/contracts/Payroll.sol

```diff
@@ -443,24 +596,62 @@ contract Payroll is AragonApp { //, IForwarder { // makes coverage crash (remove
                 1,
                 ""
             );
-            SendPayroll(employee.accountAddress, token, tokenAmount);
+            emit SendPayroll(employee.accountAddress, token, tokenAmount);
             somethingPaid = true;
         }
+
+        // try to remove employee
+        if (employees[employeeId].terminated &&
+            employees[employeeId].endDate <= getTimestamp64() &&
+            employees[employeeId].accruedValue == 0
+        ) {
+            delete employeeIds[employees[employeeId].accountAddress];
+            delete employees[employeeId];
+        }
+    }
+
+    function _terminateEmployee(uint128 employeeId, uint64 endDate) internal {
+        // prevent past termination dates
+        require(endDate >= getTimestamp64());
+
+        employees[employeeId].terminated = true;
+        employees[employeeId].endDate = endDate;
+
+        emit TerminateEmployee(employeeId, employees[employeeId].accountAddress, endDate);
+    }
+
+    function _getOwedSalary(uint128 employeeId, uint64 date) internal view returns (uint256) {
+        // get time that has gone by (seconds)
+        uint64 time = date.sub(employees[employeeId].lastPayroll);
+        if (time == 0) {
+            return 0;
+        }
+
+        return employees[employeeId].denominationTokenSalary.mul(time);
     }
 
     /**
-     * @dev Check that a token has the exchange rate already set
-     *      Internal function, needed to ensure that we have the rate before making a payment.
+     * @dev Check that a token gets a correct exchange rate.
+     *      Internal function, needed to ensure that we have a recent rate before making a payment.
      * @param token The token address
      * @return True if we have the exchange rate, false otherwise
```

Comment:
The `@return` is wrong: if we have the exchange rate, we return `ONE`, and `0` otherwise.